### PR TITLE
Add `QuestionComponent`

### PR DESCRIPTION
**Pull request by pixeldesu**
_Requires #1403 to be merged first._

De-duplicates question displays across the application.

**Note:** Does not include any tests because...

1. `ApplicationComponent` is just a base component exposing Devise user helpers to the components
2. The above is making it exceptionally hard to unit test components that require user context, because Devise in the end requires Warden/controller context in tests and that's not a given.

I'll try to find a solution for the testing problem later.

### CODE REVIEW

File: app/components/question_component.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+class QuestionComponent < ApplicationComponent
+  include ApplicationHelper
+  include BootstrapHelper
+  include UserHelper
+
+  def initialize(question:, context_user: nil, collapse: true, hide_avatar: false, profile_question: false)
+    @question = question
+    @context_user = context_user
+    @collapse = true
```
Comment (nilsding) [inline app/components/question_component.rb]: oh yeah rubogarda is right about this one

```suggestion
    @collapse = collapse
```

I wish Ruby had something like Crystal's initialize syntax, where all this wouldn't be necessary:
```crystal
def initialize(@question : Question, @context_user : User?, @collapse = true, @hide_avatar = false, @profile_question = false)
```

File: app/components/question_component/question_component.en.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+en:
+  answers:
+    zero: "0 answers"
```
Comment (raccube) [inline app/components/question_component/question_component.en.yml:3]: Could put something like "No answers" or "Unanswered" here.

**Review by raccube — APPROVED** [review]

File: app/components/question_component/question_component.en.yml (hunk 2 above)
Comment (nilsding) [reply app/components/question_component/question_component.en.yml:3]: Zarro answoors found

Comment (pixeldesu) [reply app/components/question_component/question_component.en.yml:3]: This technically doesn't appear anyway (answer count must be positive to show uppp), it's just added to match all numerical translation values

Comment (nilsding) [reply app/components/question_component/question_component.en.yml:3]: then i would suggest to change it to something more fun

```suggestion
    zero: "zefix, do is jo nix"
```

**Review by nilsding — APPROVED** [review]

**Pull request merged**